kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -395,24 +397,41 @@ public D createOrReplaceWithNew() throws KubernetesClientException {
 
   @Override
   public T createOrReplace(T... items) {
-    T item = getItem();
+    T itemToCreateOrReplace = getItem();
     if (items.length > 1) {
       throw new IllegalArgumentException("Too many items to create.");
     } else if (items.length == 1) {
-      item = items[0];
+      itemToCreateOrReplace = items[0];
     }
 
-    if (item == null) {
+    if (itemToCreateOrReplace == null) {
       throw new IllegalArgumentException("Nothing to create.");
     }
 
-    if (Utils.isNullOrEmpty(name) && item instanceof HasMetadata) {
-      return withName(((HasMetadata)item).getMetadata().getName()).createOrReplace(item);
+    if (Utils.isNullOrEmpty(name)) {
+
+      return withName(itemToCreateOrReplace.getMetadata().getName()).createOrReplace(itemToCreateOrReplace);
     }
-    if (fromServer().get() == null) {
-      return create(item);
-    } else {
-      return replace(item);
+
+    String resourceVersion = KubernetesResourceUtil.getResourceVersion(itemToCreateOrReplace);
+    try {
+      // Create
+      KubernetesResourceUtil.setResourceVersion(itemToCreateOrReplace, null);
+      return create(itemToCreateOrReplace);
+    } catch (KubernetesClientException exception) {
+      if (exception.getCode() != HttpURLConnection.HTTP_CONFLICT) {
+        throw exception;
+      }
+
+      // Conflict; Do Replace
+      T itemFromServer = fromServer().get();
+      if (ResourceCompare.equals(itemFromServer, itemToCreateOrReplace)) {
+        // Nothing changed, ignore
+        return itemToCreateOrReplace;
+      } else {
+        KubernetesResourceUtil.setResourceVersion(itemToCreateOrReplace, resourceVersion);
```

Comment:
Are we sure we want to reuse the resourceVersion retrieved in line 416? Or maybe use the one of the latest version retrieved from the server (427)?